### PR TITLE
Fix: ClickType IDs and Visitor Reward Warning

### PR DESCRIPTION
**Pull request by Obsidianninja11**
## What
Fix clicktypes, and related features such as visitor color stuff not working quite right. (sort of continuation of #1595 and [bug report](https://discord.com/channels/997079228510117908/1233943553529610320))
Also did some cleanup in VisitorRewardWarning

## Changelog Fixes
+ Fixed visitor highlight color sometimes being wrong. - Obsidian

## Changelog Technical Details
+ Fixed ClickTypeEnum ids starting at 1. - Obsidian

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/garden/visitor/VisitorRewardWarning.kt [hunk 1]
```diff
@@ -65,14 +68,17 @@ class VisitorRewardWarning {
...
             VisitorAPI.changeStatus(visitor, VisitorAPI.VisitorStatus.REFUSED, "refused")
+            // fallback if tab list is disabled
+            DelayedRun.runDelayed(10.seconds) {
+                VisitorAPI.removeVisitor(visitor.visitorName)
+            }
             return
         }
-        if (isAcceptSlot) {
-            if (stack.name != "§eClick to give!") return
+        if (isAcceptSlot && stack.getLore().contains("§eClick to give!")) {
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/garden/visitor/VisitorRewardWarning.kt:81]: why do we check the lore? the name should always have the name. this feels like a downgrade in terms for precision and performance

**Review by hannibal002 — CHANGES_REQUESTED** [review]
Good catch finding the big code duplication here and removing so much code!
The changelogs suggests the ClickType name problem this was a user facing bug. we were checking the correct ids before. this is just a internal bug/code cleanup imo.

File: src/main/java/at/hannibal2/skyhanni/features/garden/visitor/VisitorRewardWarning.kt (hunk 1 above)
Comment (Obsidianninja11) [reply src/main/java/at/hannibal2/skyhanni/features/garden/visitor/VisitorRewardWarning.kt:81]: Click to give isn't in the name
its only in the lore

```
display name: '§aAccept Offer'
lore:
 '§7Items Required:'
 ' §aCompost §8x3'
 ''
 '§7Rewards:'
 ' §8+§315k §7Farming XP'
 ' §8+§215 §7Garden Experience'
 ' §8+§c31 Copper'
 ' §8+§b10 Bits'
 ' §9Biofuel'
 ''
 '§eClick to give!'```

Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/garden/visitor/VisitorRewardWarning.kt:81]: so the original logic was useless/wrong? :O

**Review by hannibal002 — APPROVED** [review]
worked for me

**Pull request merged**